### PR TITLE
Code value trim

### PR DESCRIPTION
**Pull request by scsewall**
The Code constructor now trims leading and trailing whitespace from Code.value

### CODE REVIEW

File: core/common/src/Code.ts [hunk 1]
```diff
@@ -34,16 +34,16 @@ export class Code implements CodeProps {
   /** the [CodeValue]($docs/bis/intro/codes.md#codevalue-property) of the Element */
   public value?: string;
 
-  constructor(val: CodeProps) {
-    this.spec = Id64.fromJSON(val.spec);
-    this.scope = JsonUtils.asString(val.scope, "");
-    this.value = JsonUtils.asString(val.value);
+  constructor(codeProps: CodeProps) {
+    this.spec = Id64.fromJSON(codeProps.spec);
```
Comment (pmconne) [inline core/common/src/Code.ts:38]: Should make the public `value` a property and trim input in setter?

File: core/common/src/Code.ts [hunk 2]
```diff
@@ -34,16 +34,16 @@ export class Code implements CodeProps {
...
+    this.scope = JsonUtils.asString(codeProps.scope);
+    this.value = JsonUtils.asString(codeProps.value?.trim());
   }
 
   /** Create an empty, non-unique code with no special meaning. */
   public static createEmpty(): Code { const id: Id64String = Id64.fromLocalAndBriefcaseIds(1, 0); return new Code({ spec: id, scope: id }); }
   public static fromJSON(json?: any): Code { return json ? new Code(json) : Code.createEmpty(); }
-  public getValue(): string { return this.value ? this.value : ""; }
+  public getValue(): string { return this.value ?? ""; }
```
Comment (kabentley) [inline core/common/src/Code.ts:46]: i agree with Paul, value should be a property. Then, deprecate this method.

File: core/common/src/Code.ts (hunk 1 above)
Comment (scsewall) [reply core/common/src/Code.ts:42]: Done

File: core/common/src/Code.ts (hunk 2 above)
Comment (scsewall) [reply core/common/src/Code.ts]: Done

**Review by kabentley — APPROVED** [review]

File: core/common/src/Code.ts (hunk 1 above)
Comment (swwilson-bsi) [reply core/common/src/Code.ts:42]: Why the question mark after val in the setter?

**Review by swwilson-bsi — APPROVED** [review]

File: core/common/src/Code.ts (hunk 1 above)
Comment (scsewall) [reply core/common/src/Code.ts:42]: It seemed better to have `undefined` clear rather than throw an error.

**Pull request merged**